Missing autoloads

lafrenierejm opened the pull request:
When packaging 0.2.7 for GNU Guix, I the following warning when byte-compiling `wucuo.el`:

``` text
In end of data:
wucuo.el:662:1:Warning: the following functions are not known to be defined:
    wucuo-flyspell-html-verify, wucuo-flyspell-org-verify,
    markdown-flyspell-check-word-p,
    wucuo-flyspell-highlight-incorrect-region-hack
```

I believe this is all a matter of missing autoload declarations.

redguardtoo commented:
cd8b0eb some autoload warning fix (Chen Bin)

Not sure it's fixed or not. Actually byte compiling is fine on my side.

lafrenierejm commented:
0.2.8 appears to have resolved the warning for `wucuo-flyspell-highlight-incorrect-region-hack`, but I'm still getting warnings for the other functions:

```text
wucuo.el:775:1:Warning: the following functions are not known to be defined:
    wucuo-flyspell-html-verify, wucuo-flyspell-org-verify,
    markdown-flyspell-check-word-p
```

lafrenierejm commented:
Looking at `wucuo-flyspell-html-verify`, I see that
1. The function is defined in `wucuo-flyspell-html-verify.el`.
1. The function is referenced in `wucuo.el`.
1. `wucuo.el` never `require`s `wucuo-flyspell-html-verify`.

lafrenierejm committed:
require wucuo-flyspell-{html,org}-verify

lafrenierejm commented:
https://github.com/lafrenierejm/wucuo/commit/ef9801d7e77f8404642cacad88f3c5d162646dd4 resolved the issue for `wucuo-flyspell-html-verify` and `wucuo-flyspell-org-verify`.  The warning is still present for `markdown-flyspell-check-word-p`;  I need to take a closer look through the code before I'll have a suggestion for that function.

lafrenierejm committed:
register markdown-mode autoload at top level

lafrenierejm committed:
remove duplicate autoloads

The autoloading of the wucuo functions occurs in their defining packages.  The
markdown function does not need to be autoloaded.

lafrenierejm commented:
I have updated the branch https://github.com/lafrenierejm/wucuo/tree/bytecompile-warning.  With the current HEAD (89b99166768afb811c48a7db7c93c02d51a32b09), all of the autoload warnings are resolved.

redguardtoo commented:
I try to avoid `require` for performance reason. Let me figure out a solution

lafrenierejm commented:
Everything looked good when packaging 0.2.9, so I'll close this.  Thanks for the package!
